### PR TITLE
OKTA-459959 Adding the Embedded Sign-In: SPA with Embedded SIW - Angular

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Addition of OIE specific guide to replace the [Angular Okta Sign-In Widget Guide](https://developer.okta.com/code/angular/okta_angular_sign-in_widget/)
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* [OKTA-459959 ](https://oktainc.atlassian.net/browse/OKTA-459959 )

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 1]
```diff
@@ -1,3 +1,133 @@
-## Angular
+## About Angular and the Okta Sign-In Widget
 
-To come ....
+The embedded Okta Sign-In Widget provides user authentication, as well as an opportunity to customize the sign in experience for users of your Angular application.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:3]:
```suggestion
The embedded Okta Sign-In Widget provides user authentication, as well as an opportunity to customize the sign-in experience for users of your Angular application.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 2]
```diff
@@ -1,3 +1,133 @@
...
+
+This guide explains how to build a sample application with the embedded Sign-In Widget included in the [okta/samples-js-angular](https://github.com/okta/samples-js-angular) repository. See [Create a simple authentication use case with Angular](#create-a-simple-authentication-use-case-with-angular) for further walk-through instructions on how to implement this sample application.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:5]:
```suggestion
This guide explains how to build a sample application with the embedded Sign-In Widget that is included in the [okta/samples-js-angular](https://github.com/okta/samples-js-angular) repository. See [Create a simple authentication use case with Angular](#create-a-simple-authentication-use-case-with-angular) for further walk-through instructions on how to implement this sample application.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 3]
```diff
@@ -1,3 +1,133 @@
...
+
+<img src="/img/okta-sign-in-angular.png" alt="Screenshot of basic Okta Sign-In Widget using Angular" width="800">
+
+## Installation
+
+For all applications, the first step is to install the Widget. You have two options: linking out to the Okta CDN, or installing locally through `npm`.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:11]:
```suggestion
For all applications, the first step is to install the Widget. You have two options: linking out to the Okta CDN or installing locally through `npm`.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 4]
```diff
@@ -1,3 +1,133 @@
...
+
+To use the CDN, include the following script in your HTML:
+
+```html
+<!-- Latest CDN production JavaScript and CSS -->
+<script src="https://global.oktacdn.com/okta-signin-widget/-=OKTA_REPLACE_WITH_WIDGET_VERSION=-/js/okta-sign-in.min.js" type="text/javascript"></script>
+<link href="https://global.oktacdn.com/okta-signin-widget/-=OKTA_REPLACE_WITH_WIDGET_VERSION=-/css/okta-sign-in.min.css" type="text/css" rel="stylesheet"/>
+```
+
+More info, including the latest published version, can be found in the [Widget Documentation](https://github.com/okta/okta-signin-widget#using-the-okta-cdn).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:23]:
```suggestion
More info, including the latest published version, can be found in the [Widget documentation](https://github.com/okta/okta-signin-widget#using-the-okta-cdn).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 5]
```diff
@@ -1,3 +1,133 @@
...
+
+### npm
+
+```bash
+# Run this command in your project root folder.
+npm install @okta/okta-signin-widget@-=OKTA_REPLACE_WITH_WIDGET_VERSION=-
+```
+
+More info, including the latest published version, can be found in the [Widget Documentation](https://github.com/okta/okta-signin-widget#using-the-npm-module).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:32]:
```suggestion
More info, including the latest published version, can be found in the [Widget documentation](https://github.com/okta/okta-signin-widget#using-the-npm-module).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 6]
```diff
@@ -1,3 +1,133 @@
...
+
+## Create a simple authentication use case with Angular
+
+In this use case, you use the Widget to sign in to a simple web page that provides details on the user signed in. Ensure you have an Okta developer account, and use the github repo sample.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:36]: Provide a link to the github repo sample?

```suggestion
In this use case, you use the Widget to sign in to a simple web page that provides details on the user signed in. Ensure that you have an Okta developer account and use the github repo sample.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 7]
```diff
@@ -1,3 +1,133 @@
...
+
+To create and run this sample use case:
+
+* Create an app integration on your Okta org.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:40]:
```suggestion
* Create an app integration in your Okta org.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 8]
```diff
@@ -1,3 +1,133 @@
...
+* Download the simple SPA from the gitHub repository.
+* Run the sample application.
+
+>**Note:** This use case uses Angulr x, Node y, and Auth-JS blah ******TBD*******
+
+### Create the app integration
+
+Create an app integration that represents the Angular application you want to add authentication to using Okta:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:48]:
```suggestion
Create an app integration that represents the Angular application that you want to add authentication to using Okta:
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 9]
```diff
@@ -1,3 +1,133 @@
...
+
+1. In the Admin Console, go to **Applications** > **Applications**.
+1. Click **Create App Integration**.
+1. Select **OIDC - OpenID Connect** as the **Sign-in method**.
+1. Select **Single-Page Application** for the **Application Type**.
+1. On the **New Single-Page App Integration** page:
+
+   * Enter an application name.
+   * Select the **Interaction Code** check box.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:57]:
```suggestion
   * Select the **Interaction Code** checkbox.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 10]
```diff
@@ -1,3 +1,133 @@
...
+   * Select the **Refresh Token** check box.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:58]:
```suggestion
   * Select the **Refresh Token** checkbox.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 11]
```diff
@@ -1,3 +1,133 @@
...
+   * Set **Base URI** to `http://localhost:8080`
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:59]:
```suggestion
   * Set **Base URI** to `http://localhost:8080`.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 12]
```diff
@@ -1,3 +1,133 @@
...
+   * Set **Sign-in redirect URIs** to `http://localhost:8080/login/callback`.
+   * Set **Sign-out redirect URIs** to `http://localhost:8080/`
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:61]:
```suggestion
   * Set **Sign-out redirect URIs** to `http://localhost:8080/`.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 13]
```diff
@@ -1,3 +1,133 @@
...
+
+1. In the **Assignments** section, select **Allow everyone in your organization to access**.
+1. Click **Save**.
+1. Select the **Sign On** tab.
+1. In the **Sign On Policy** section, verify that the **Available Authenticators** settings are appropriate for your app. For this use case, ensure that the **1 factor type** authenticator is **Password / IdP**.
+1. In the **Security** > **API** > **Authorization Servers** section, verify that the custom authorization server uses the Interaction Code grant type by selecting the **default** server, clicking **Access Policies**, and editing the **Default Policy Rule**. Review the **If Grant type is** section to ensure the **Interaction Code** check box is selected.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:67]:
```suggestion
1. In the **Security** > **API** > **Authorization Servers** section, verify that the custom authorization server uses the Interaction Code grant type by selecting the **default** server, clicking **Access Policies**, and editing the **Default Policy Rule**. Review the **If Grant type is** section to ensure the **Interaction Code** checkbox is selected.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 14]
```diff
@@ -1,3 +1,133 @@
...
+1. In the **Security** > **API** > **Trusted Origins** page, ensure there is an entry for your sign in redirect URI. See [Enable CORS](/docs/guides/enable-cors/).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:68]:
```suggestion
1. In the **Security** > **API** > **Trusted Origins** page, ensure that there is an entry for your sign-in redirect URI. See [Enable CORS](/docs/guides/enable-cors/).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 15]
```diff
@@ -1,3 +1,133 @@
...
+
+> **Note:** From the **General** tab of your app integration, save the generated **Client ID** value, which is used in the next section.
+
+### Download the sample Angular application
+
+1. In your project folder, download the sample application from the gitHub repository, and navigate to the custom-login folder.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:74]:
```suggestion
1. In your project folder, download the sample application from the gitHub repository and navigate to the custom-login folder.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 16]
```diff
@@ -1,3 +1,133 @@
...
+    cd samples-js-angular/custom-login
+    ```
+
+1. In the `custom-login` folder, install the dependencies:
+
+    ```bash
+    npm install
+    ```
+
+1. Ensure you're using the angular version and node version designed for this sample application.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:87]:
```suggestion
1. Ensure that you're using the angular version and node version designed for this sample application.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 17]
```diff
@@ -1,3 +1,133 @@
...
+    ng --version
+    ```
+
+1. Install the following version of `okta-auth-js` SDK:
+
+    ```bash
+    npm install @okta/okta-auth-js@5.9.1
+    ```
+
+1. Navigate to the `samples-js-angular` folder, and create a configuration file, `testenv` (no extension).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:103]:
```suggestion
1. Navigate to the `samples-js-angular` folder and create a configuration file called `testenv` (no extension).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 18]
```diff
@@ -1,3 +1,133 @@
...
+
+1. From the `custom-login` folder, run the sample application:
+
+    ```bash
+    ng serve
+    ```
+
+1. After a successful compilation, navigate to `http://localhost:8080`. The **Custom Login with Sign-In Widget** page appears.
+
+1. Click the **Login** button, and enter credentials for a user assigned to your app integration. The **Custom Login with Sign-In Widget** returns with the user's email address.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:125]:
```suggestion
1. Click **Login** and enter credentials for a user assigned to your app integration. The **Custom Login with Sign-In Widget** returns with the user's email address.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 19]
```diff
@@ -1,3 +1,133 @@
...
+
+1. Click the **Profile** tab to view claim information returned in the ID token.
+
+## Next steps
+
+You have now successfully authenticated with Okta! Now what? With a user's `id_token`, you have basic claims for the user's identity. You can extend the set of claims by modifying the `scopes` to retrieve custom information about the user. This includes `locale`, `address`, `groups`, and [more](/docs/reference/api/oidc/).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:131]:
```suggestion
You have now successfully authenticated with Okta. Now what? With a user's `id_token`, you have basic claims for the user's identity. You can extend the set of claims by modifying the `scopes` to retrieve custom information about the user. This includes `locale`, `address`, `groups`, and [more](/docs/reference/api/oidc/).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md [hunk 20]
```diff
@@ -1,3 +1,133 @@
...
+
+Want to learn how to use the user's `access_token`? Check out our [how-to guide](/docs/guides/sign-into-spa/angular/main/) to learn about protecting routes on your server, validating the `access_token`, and more!
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/guide.md:133]:
```suggestion
Want to learn how to use the user's `access_token`? Check out our [how-to guide](/docs/guides/sign-into-spa/angular/main/) to learn about protecting routes on your server, validating the `access_token`, and more.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/nutrition.md [hunk 21]
```diff
@@ -1,22 +1,21 @@
-This guide explains how to sign in to your SPA with the Okta Sign-In Widget.
+This guide explains how to sign in to an Angular framework single page application (SPA) with the embedded Okta Sign-In Widget.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/nutrition.md:1]:
```suggestion
This guide explains how to sign in to an Angular framework single-page application (SPA) with the embedded Okta Sign-In Widget.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/nutrition.md [hunk 22]
```diff
@@ -1,22 +1,21 @@
...
 ---
 
 **Learning outcomes**
 
-* Angular - TBD
-* Angular - TBD
-* Angular - TBD
+* Understand how to implement a sign-in use case using Angular and the Okta Sign-In Widget.
+* Understand basic installation and code configurations using Angular.
+* Implement a simple SPA use-case, and sign a user in to the application.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/nutrition.md:9]:
```suggestion
* Implement a simple SPA use-case and sign a user in to the application.
```

**Review by susanharper-okta — COMMENTED** [review]
couple nits - style

**Comment by brianduffield-okta** [comment]
Closing this PR; content was ported over to PR https://github.com/okta/okta-developer-docs/pull/2976 and uses the new branch nav structure.

**Pull request closed without merging**